webview/App.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useRef, useState, useCallback } from 'react'
+import { throttle } from './utils'
+
+declare const acquireVsCodeApi: any
+
+interface ReplOutputItem {
+    type: string
+    text: string
+    pkgName?: string
+}
+
+export default function AliveREPL() {
+    const [vscodeApi, setVscodeApi] = useState<any>(null)
+
+    const [packageName, setPackageName] = useState('')
+    const [inputText, setInputText] = useState('')
+    const [outputItems, setOutputItems] = useState<ReplOutputItem[]>([])
+
+    const scrollReplRef = useRef<Function>(null)
+    const inputRef = useRef<HTMLInputElement>(null)
+    const replOutputRef = useRef<HTMLDivElement>(null)
+    const calledOnce = useRef(false)
+
+    const messageHandler = useCallback((event: MessageEvent) => {
+        const data = event.data;
+
+        switch (data.type) {
+            case 'setInput':
+                setInputText(data.text);
+                break;
+            case 'clearInput':
+                setInputText('');
+                break;
+            case 'setOutput':
+                setOutputItems(data.items);
+                scrollReplRef.current?.();
+                break;
+            case 'appendOutput':
+                setOutputItems(prevOutputItems => [...prevOutputItems, data.obj]);
+                scrollReplRef.current?.();
+                break;
+            case 'clear':
+                setOutputItems([]);
+                break;
+            case 'setPackage':
+                setPackageName(data.name);
+                inputRef.current?.focus();
+                break;
+            default:
+                break;
+        }
+    }, []); // Empty array makes the handler stable
+
+    useEffect(() => {
+        if (calledOnce.current) return
+        calledOnce.current = true
+
+        // VSCode instance can be acquired only once
+        setVscodeApi(acquireVsCodeApi())
+
+        // Scroll the output view with a delay
+        scrollReplRef.current = throttle(() => scrollReplView(), 80)
+
+        // Get initial package name and extension version from root element
+        const rootElement = document.getElementById('root');
+        if (rootElement) {
+            const initPackage = rootElement.getAttribute('data-init-package')
+            if (initPackage) {
+                setPackageName(initPackage)
+            }
+
+            const extensionVersion = rootElement.getAttribute('data-extension-version')
+            if (extensionVersion) {
+                console.log("inside")
+                setOutputItems((prevOutputItems) => {
+                    const versionInfo = `; Alive REPL (v${extensionVersion})`
```

Comment:
This is duplicated in extension.ts, but it seems like this one might not be getting added to the REPL output list.